ui/src/criteria/attribute.tsx

```diff
@@ -0,0 +1,153 @@
+import { Checkbox, FormControlLabel, ListItem } from "@mui/material";
+import Stack from "@mui/material/Stack";
+import Typography from "@mui/material/Typography";
+import { CriteriaConfig, CriteriaPlugin, registerCriteriaPlugin } from "cohort";
+import produce from "immer";
+import React from "react";
+import * as tanagra from "tanagra-api";
+import { useUnderlay } from "../hooks";
+
+type Selection = {
+  id: number;
+  name: string;
+};
+
+interface Config extends CriteriaConfig {
+  attribute: string;
+}
+
+interface Data extends Config {
+  selected: Selection[];
+}
+
+type AttributeEditProps = {
+  dispatchFn: (data: Data) => void;
+  data: Data;
+};
+
+@registerCriteriaPlugin("attribute", (config: CriteriaConfig) => ({
+  ...(config.plugin as Config),
+  selected: [],
+}))
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+class _ implements CriteriaPlugin<Data> {
+  public data: Data;
+
+  constructor(public id: string, data: unknown) {
+    this.data = data as Data;
+  }
+
+  renderEdit(dispatchFn: (data: Data) => void) {
+    return <AttributeEdit dispatchFn={dispatchFn} data={this.data} />;
+  }
+
+  renderDetails() {
+    return <ConceptDetails data={this.data} />;
+  }
+
+  generateFilter() {
+    if (this.data.selected.length === 0) {
+      return null;
+    }
+
+    const operands = () =>
+      this.data.selected.map(({ id }) => ({
+        binaryFilter: {
+          attributeVariable: {
+            variable: "person",
+            name: this.data.attribute,
+          },
+          operator: tanagra.BinaryFilterOperator.Equals,
+          attributeValue: {
+            int64Val: id,
+          },
+        },
+      }));
+
+    return {
+      arrayFilter: {
+        operands: operands(),
+        operator: tanagra.ArrayFilterOperator.Or,
+      },
+    };
+  }
+
+  occurrenceEntities() {
+    return [];
+  }
+}
+
+function AttributeEdit(props: AttributeEditProps) {
+  const underlay = useUnderlay();
+  const attributeName = props.data.attribute;
```

Comment:
You don't have to change this if you like it this way but this is only a few characters shorter. Introducing a new variable makes the code more complicated since you have to go back and check what was assigned to it. It makes more sense to do that when it removes duplicated logic or makes the code notably smaller or easier to read.